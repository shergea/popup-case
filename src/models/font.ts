export default interface Font {
  entities: Array<FontItem>;
  loading: "idle" | "pending" | "succeeded" | "failed";
  expireTime: number;
}

export interface FontItem {
  family: string;
  subsets: string[];
  version: string;
  category: string;
  variants: string[];
}
