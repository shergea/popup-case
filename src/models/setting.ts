export default interface Setting {
  headline: string;
  description: string;
  successMessage: string;
  buttonText: string;
}
