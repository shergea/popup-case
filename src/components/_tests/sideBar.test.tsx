import React from "react";
import renderer from "react-test-renderer";
import { FontItem } from "../../models/font";
import Setting from "../../models/setting";
import Sidebar from "../sideBar";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Sidebar
        onFormChange={() => {}}
        data={
          {
            headline: "HeadLine",
            buttonText: "ButtonText",
            description: "Description",
            successMessage: "successMessage",
            font: "Roboto",
          } as Setting
        }
        fonts={
          [
            {
              family: "ABeeZee",
              subsets: ["latin", "latin-ext"],
              version: "v22",
              category: "sans-serif",
              variants: ["regular", "italic"],
            },
            {
              family: "Abel",
              subsets: ["latin"],
              version: "v18",
              category: "sans-serif",
              variants: ["regular"],
            },
          ] as Array<FontItem>
        }
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
