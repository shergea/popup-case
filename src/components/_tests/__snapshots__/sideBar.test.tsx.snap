// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col h-screen p-16 bg-white shadow"
>
  <h1
    className="text-2xl font-semibold underline"
  >
    General Settings
  </h1>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="mt-5"
    >
      <label
        className="block mb-2 text-m font-medium text-gray-900 dark:text-gray-300"
        htmlFor="headline"
      >
        Headline
      </label>
      <input
        className="border border-gray-300 text-gray-900 text-m rounded-lg block w-full p-2.5 outline-0"
        id="headline"
        name="headline"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="NEW STUFF"
        type="text"
        value="HeadLine"
      />
    </div>
    <div
      className="mt-5"
    >
      <label
        className="block mb-2 text-m font-medium text-gray-900 dark:text-gray-300"
        htmlFor="description"
      >
        Description
      </label>
      <textarea
        className="border border-gray-300 text-gray-900 text-m rounded-lg block w-full p-2.5 resize-none outline-0"
        id="description"
        name="description"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Description"
        rows={5}
        value="Description"
      />
    </div>
    <div
      className="mt-5"
    >
      <label
        className="block mb-2 text-m font-medium text-gray-900 dark:text-gray-300"
        htmlFor="buttonText"
      >
        Button Text
      </label>
      <input
        className="border border-gray-300 text-gray-900 text-m rounded-lg block w-full p-2.5 outline-0"
        id="buttonText"
        name="buttonText"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Button Text"
        type="text"
        value="ButtonText"
      />
    </div>
    <div
      className="mt-5"
    >
      <label
        className="block mb-2 text-m font-medium text-gray-900 dark:text-gray-300"
        htmlFor="successMessage"
      >
        Success Message
      </label>
      <input
        className="border border-gray-300 text-gray-900 text-m rounded-lg block w-full p-2.5 outline-0"
        id="successMessage"
        name="successMessage"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Success"
        type="text"
        value="successMessage"
      />
    </div>
    <div
      className="mt-4"
    >
      <label
        className="block mb-2 text-m font-medium text-gray-900 dark:text-gray-300"
        htmlFor="fontSelector"
      >
        Font
      </label>
      <select
        className="border border-gray-300 text-gray-900 text-m rounded-lg block w-full p-2.5 outline-0"
        id="fontSelector"
        name="font"
        onBlur={[Function]}
        onChange={[Function]}
        value="Roboto"
      >
        <option
          value=""
        >
          Select Font
        </option>
        <option
          value="'ABeeZee', sans-serif"
        >
          ABeeZee
        </option>
        <option
          value="'Abel', sans-serif"
        >
          Abel
        </option>
      </select>
    </div>
  </form>
</div>
`;
